keps/sig-node/1029-ephemeral-storage-quotas/README.md

```diff
@@ -576,6 +579,79 @@ required elsewhere:
+managing filesystem quotas within containers. The toggle allows 
+to switch between a quota management system and a traditional 
+disk usage monitoring method.
+
+* During container creation, there must be a check to verify 
+if xfs-quota in Kubelet is enabled (whether 
+LocalStorageCapacityIsolationFSQuotaMonitoring flag is set to true 
+or not). Additionally, verify if hostUsers in pod is set to false,
+indicating that user namespaces are enabled.
+
+* Only if both, xfs-quota and user namespace, are enabled then 
+the toggle enables xfs-quota on the container. If either xfs-quota 
+is not enabled or hostUsers is set to true, then the kubelet should 
+fallback to the legacy approach of using du and find commands to 
+periodically gather information about storage and inode consumption 
+of volumes
+
+### Risks and Mitigations
+
+* The SIG raised the possibility of a container being unable to exit
+  should we enforce quotas, and the quota interferes with writing the
+  log.  This can be mitigated by either not applying a quota to the
+  log directory and using the du mechanism, or by applying a separate
+  non-enforcing quota to the log directory.
+
+  As log directories are write-only by the container, and consumption
+  can be limited by other means (as the log is filtered by the
+  runtime), I do not consider the ability to write uncapped to the log
+  to be a serious exposure.
+
+  Note in addition that even without quotas it is possible for writes
+  to fail due to lack of filesystem space, which is effectively (and
+  in some cases operationally) indistinguishable from exceeding quota,
+  so even at present code must be able to handle those situations.
+
+* Filesystem quotas may impact performance to an unknown degree.
+  Information on that is hard to come by in general, and one of the
+  reasons for using quotas is indeed to improve performance.  If this
+  is a problem in the field, merely turning off quotas (or selectively
+  disabling project quotas) on the filesystem in question will avoid
+  the problem.  Against the possibility that cannot be done
+  (because project quotas are needed for other purposes), we should
+  provide a way to disable use of quotas altogether via a feature
+  gate.
+
+  A report <https://blog.pythonanywhere.com/110/> notes that an
+  unclean shutdown on Linux kernel versions between 3.11 and 3.17 can
+  result in a prolonged downtime while quota information is restored.
+  Unfortunately, [the link referenced
+  here](http://oss.sgi.com/pipermail/xfs/2015-March/040879.html) is no
+  longer available.
+
+* Bugs in the quota code could result in a variety of regression
+  behavior.  For example, if a quota is incorrectly applied it could
+  result in ability to write no data at all to the volume.  This could
+  be mitigated by use of non-enforcing quotas.  XFS in particular
+  offers the `pqnoenforce` mount option that makes all quotas
+  non-enforcing.
+
+
+## Design Details
+
 ### Test Plan
```

Comment:
Probably worth updating the unit test coverage, they are 2 years old.

Reply:
Any particular reason why we no integration tests are planned?

Reply:
kubelet doesn't really define many integration tests, going for unit and then node e2e instead